src/Action/ApiDocsAction.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace PhpSchool\Website\Action;
+
+use PhpSchool\Website\DocGenerator;
+use PhpSchool\Website\PhpRenderer;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Http\Message\ResponseInterface as Response;
+
+/**
+ * Class ApiDocsAction
+ * @package PhpSchool\Website
+ * @author Aydin Hassan <aydin@hotmail.co.uk>
+ */
+class ApiDocsAction
+{
+
+    /**
+     * @var PhpRenderer
+     */
+    private $renderer;
+
+    /**
+     * @var DocGenerator
+     */
+    private $docGenerator;
+
+    /**
+     * @var CacheItemPoolInterface
+     */
+    private $cache;
+
+    public function __construct(PhpRenderer $renderer, DocGenerator $docGenerator, CacheItemPoolInterface $cache)
+    {
+        $this->renderer = $renderer;
+        $this->docGenerator = $docGenerator;
+        $this->cache = $cache;
+    }
+
+    public function __invoke(Request $request, Response $response) : Response
+    {
+        $namespace  = $request->getAttribute('route')->getArgument('namespace');
+        $class      = $request->getAttribute('route')->getArgument('class');
+
+        $apiDocs = $this->cache->getItem('api-docs');
```

Comment:
does this always return an obj that has the `isHit` method?

probably a stupid question because if not I guess you'd have seen a problem pretty qiuckly 😂